src/hashTable.js

```diff
@@ -0,0 +1,64 @@
+export default class HashTable {
+  constructor() {
+    this.hashTable = []
+    this.count = 0
+  }
+
+  hash( key ) {
+    let sum = 0;
+    const randomH = 31
+    const prime = 51
+    key = key.toString();
+    for (let i = 0; i < key.length; i++) {
+        sum = (randomH * sum) + key.charCodeAt(i)
+    }
+    return sum % prime // return hashingIndex
+  }
+
+  put( key, value ) {
+    const hashedKey = this.hash( key )
+    this.hashTable[hashedKey] = { 'key': key, 'value':value }
+    this.count++
+  }
+
+  get( key ) {
+    const hashedKey = this.hash( key )
+    const value = this.hashTable[hashedKey].value
+    if( value == undefined ) {
+      throw new Error('Key does not exist')
+    }
+    return value
+  }
+
+  contains( key )  {
+    const hashedKey = this.hash( key )
+    const value = this.hashTable[hashedKey]
+    return Boolean(value)
```

Comment:
This code feels a little too clever for me - I would try to make this as explicit and intention-revealing as possible:
```js
return this.hashTable[ this.hash( key ) ] !== undefined
```